Mathlib/NumberTheory/NumberField/Discriminant.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2023 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import Mathlib.NumberTheory.NumberField.Basic
+import Mathlib.RingTheory.Localization.NormTrace
+
+/-!
+# Number field discriminant
+This file defines the discriminant of a number field.
+
+## Main definitions
+ - `discr` the absolute discriminant of a number field.
+
+## Tags
+number field, discriminant
+-/
+
+-- TODO. Rewrite some of the FLT results on the disciminant using the definitions and results of
+-- this file
+
+namespace NumberField
+
+open NumberField Matrix
+
+variable (K : Type*) [Field K] [NumberField K]
+
+/-- The absolute discriminant of a number field. -/
+noncomputable abbrev discr : ℤ := Algebra.discr ℤ (RingOfIntegers.basis K)
+
+theorem coe_discr : (discr K : ℚ) = Algebra.discr ℚ (integralBasis K) :=
+  (Algebra.discr_localizationLocalization ℤ _ K (RingOfIntegers.basis K)).symm
+
+theorem discr_ne_zero : discr K ≠ 0 := by
+  rw [← (Int.cast_injective (α := ℚ)).ne_iff, coe_discr]
+  exact Algebra.discr_not_zero_of_basis ℚ (integralBasis K)
+
+theorem discr_eq_discr {ι : Type*} [Fintype ι] [DecidableEq ι] (b : Basis ι ℤ (𝓞 K)) :
+    Algebra.discr ℤ b = discr K := by
+  let b₀ := Basis.reindex (RingOfIntegers.basis K) (Basis.indexEquiv (RingOfIntegers.basis K) b)
+  rw [Algebra.discr_eq_discr (𝓞 K) b b₀, Basis.coe_reindex, Algebra.discr_reindex]
+
+end NumberField
+
+namespace Rat
+
+open NumberField
+
+/-- The absolute discriminant of the number field `ℚ` is 1. -/
+theorem numberField_discr : discr ℚ = 1 := by
```

Comment:
This can be a simp lemma. Would you also please alias this lemma as `NumberField.discr_rat`?